src/pipeline-provider.ts

```diff
@@ -68,8 +83,8 @@ export class PipelineProvider {
   addStep(name: string, step: PipelineStep): void {
     let found = this._findStep(name);
     if (found) {
-      if (!found.steps.includes(step)) { // prevent duplicates
-        found.steps.push(step);
+      if (!(found as PipelineSlot).steps.includes(step)) { // prevent duplicates
+        (found as PipelineSlot).steps.push(step);
```

Comment:
unnecessary typing

Reply:
Was because `found` could be either `PipelineSlot` or `Function` from existing type definition

Reply:
Edit: It was necessary because of [this](https://github.com/aurelia/router/pull/615/files/f41a488866ce19b1e3e2bc859262665818990519#diff-e0658779b23fb0d91b6c56ce52c62769R53)

Reply:
I never was able to see what you were linking to. Do you know what its necessary?

Reply:
Somehow the old commits are gone. It's necessary because the way it works with container. If we gives it a Function, it will ask the container to create an instance of it, otherwise, as object, we assume it's an object with `PipelineSlot` shape

Reply:
For whatever reason, I was thinking `ActivateNextStep` was an actual step and not the class constructor.